pdr_backend/sim/sim_plotter.py

```diff
@@ -31,28 +34,33 @@ def __init__(
         self.st = st
         self.ppss = ppss
 
+        plt.close("all")
+
         # so labels are above lines. Must be before figure()
         plt.rcParams["axes.axisbelow"] = False
 
-        # figure, subplots
-        fig = plt.figure()
-        self.fig = fig
-
-        gs = gridspec.GridSpec(2, 6, width_ratios=[5, 1, 2, 0.6, 2, 3])
-
-        self.ax_pdr_profit_vs_time = fig.add_subplot(gs[0, 0])
-        self.ax_trader_profit_vs_time = fig.add_subplot(gs[1, 0])
-
-        self.ax_pdr_profit_vs_ptrue = fig.add_subplot(gs[0, 1])
-        self.ax_trader_profit_vs_ptrue = fig.add_subplot(gs[1, 1])
-
-        self.ax_accuracy_vs_time = fig.add_subplot(gs[0, 2])
-        self.ax_f1_precision_recall_vs_time = fig.add_subplot(gs[1, 2])
-
-        # col 3 is empty, for overflow of aimodel_varimps's y-axis labels
-        self.ax_aimodel_varimps = fig.add_subplot(gs[:, 4])
-
-        self.ax_aimodel_response = fig.add_subplot(gs[:, 5])
+        c1, c2, c3 = streamlit.columns((1, 2, 1))
+        c11, c12 = c1.container(), c1.container()
+        c31, c32 = c3.container(), c3.container()
+        c4, c5 = streamlit.columns((1, 1))
+
+        self.canvas = {
+            "pdr_profit_vs_time": c11.empty(),
+            "trader_profit_vs_time": c12.empty(),
+            "accuracy_vs_time": c2.empty(),
+            "pdr_profit_vs_ptrue": c31.empty(),
+            "trader_profit_vs_ptrue": c32.empty(),
+            "aimodel_varimps": c4.empty(),
+            "aimodel_response": c5.empty(),
+            "f1_precision_recall_vs_time": streamlit.empty(),
+        }
+
+        self.figs = {}
+
+        for k, _ in self.canvas.items():
+            fig, ax = plt.subplots()
+            self.figs[k] = fig
+            setattr(self, f"ax_{k}", ax)
```

Comment:
@calina-c I'm a bit confused as to why we would register the figs into self.figs[dict] and the axes[] into a `self.ax_` property. 

I.E. 1 object is being split into 2 different interfaces, is there a reason for this?

Reply:
it was just an intermediary way of dealing with multiple figures. Once we get rid of the plot figures and keep streamlit ones, there is no need to keep figures. In addition to that, the architecture will also change by implementing a data layer, so this intermediary crutch of an implementation is moot.